### PR TITLE
PWGHF: Implementing B0 → D- π+ → (π- K+ π-) π+ workflow

### PR DESCRIPTION
**Pull request by AlexBigO**
Implementation of the B0 workflow along with [this PR](https://github.com/AliceO2Group/O2Physics/pull/1282).

### CODE REVIEW

**Comment by vkucera** [comment]
Hi @AlexBigO , thanks for the PR. I see that the commit called "Clang-formatting" formats the JSON file and changes indentation from 4 spaces (which is the format dumped by O2) to 2 spaces. What tool did you use for the formatting?

**Comment by AlexBigO** [comment]
Hello again @vkucera . I used `git clang-format` while being in O2.

**Comment by vkucera** [comment]
> Hello again @vkucera . I used `git clang-format` while being in O2.

But `clang-format` is a C++ formatting tool. It doesn't format JSON.

**Comment by AlexBigO** [comment]
I just tested it again (by deleting the last commit and clang-formatting) and it is indeed this command that changes the JSON. Indeed, the clang-format is for `.cxx` and I did it automatically (by habit) before opening the PR, I did not know it would change the JSON.
Anyway, I just pushed the same content but without the "Clang-formatting" commit.

File: codeHF/dpl-config_run3.json [hunk 1]
```diff
@@ -402,6 +402,19 @@
         "minRelChi2Change": "0.9",
         "doValPlots": "true"
     },
+    "hf-candidate-creator-b0": {
+        "magneticField": "20",
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:406]: Shouldn't this be 5?

File: codeHF/dpl-config_run3.json [hunk 2]
```diff
@@ -402,6 +402,19 @@
...
+        "b_propdca": "true",
+        "d_maxr": "200",
+        "d_maxdzini": "4",
+        "d_minparamchange": "0.001",
+        "d_minrelchi2change": "0.90000000000000002",
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:411]: Truncate the values with the trailing digits.

File: codeHF/dpl-config_run3.json [hunk 3]
```diff
@@ -815,6 +828,237 @@
...
+                    "0.01",
+                    "1",
+                    "0.14999999999999999",
+                    "0.050000000000000003",
+                    "0.050000000000000003",
+                    "0",
+                    "0.10000000000000001",
+                    "0.80000000000000004"
+                ],
+                [
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:904]: Make this a proper array with rows and columns, as done in other selectors.

File: codeHF/dpl-config_run3.json [hunk 4]
```diff
@@ -1096,6 +1340,48 @@
         "cutEtaCandMax": "-1",
         "processMC": "true"
     },
+    "hf-task-b0": {
+        "selectionFlagB0": "1",
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:1344]: Selection flags should be 0 by default.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: codeHF/dpl-config_run3.json [hunk 5]
```diff
@@ -1185,6 +1273,48 @@
...
+                "13",
+                "16",
+                "20",
+                "24"
+            ]
+        }
+    },
+    "hf-task-b0-mc": {
+        "selectionFlagB0": "0",
+        "cutYCandMax": "0.80",
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:1299]: Why do you cut |y| < 0.8 here but 1.44 in `hf-task-b0`?

File: codeHF/workflows.yml [hunk 6]
```diff
@@ -74,6 +74,13 @@ workflows:
       default: [HFCANDCHICBASE, HFCANDCHICEXT]
       mc: [HFCANDCHICMCREC, HFCANDCHICMCGEN]
 
+  o2-analysis-hf-candidate-creator-b0:
+    <<: *cand_creator
+    dependencies: o2-analysis-hf-b0-todpi-candidate-selector
```
Comment (vkucera) [inline codeHF/workflows.yml:79]: Shouldn't the dependency be the D0 selector?

File: codeHF/config_tasks.sh [hunk 7]
```diff
@@ -451,6 +464,7 @@ function MakeScriptPostprocess {
     [ $DOO2_TASK_JPSI -eq 1 ] && PARTICLES+=" jpsi "
     [ $DOO2_TASK_LCK0SP -eq 1 ] && PARTICLES+=" lc-tok0sP "
     [ $DOO2_TASK_XICC -eq 1 ] && PARTICLES+=" xicc-mc "
+    [ $DOO2_TASK_B0 -eq 1 ] && PARTICLES+=" b0 "
```
Comment (vkucera) [inline codeHF/config_tasks.sh:467]: You store the MC histograms in the `b0-mc` directory, not `b0`.

**Review by vkucera — CHANGES_REQUESTED** [review]
Did you try to run with all pieces activated? It seems to me that it cannot run as it is now.

File: codeHF/workflows.yml (hunk 6 above)
Comment (vkucera) [inline codeHF/workflows.yml:79]:
```suggestion
    dependencies: o2-analysis-hf-dplus-topikpi-candidate-selector
```

File: codeHF/config_tasks.sh (hunk 7 above)
Comment (vkucera) [inline codeHF/config_tasks.sh:467]:
```suggestion
    [ $DOO2_TASK_B0 -eq 1 ] && PARTICLES+=" b0-mc "
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by vkucera — APPROVED** [review]

**Pull request merged**